internal/hcl/parser.go

```diff
@@ -553,10 +562,16 @@ func (p *Parser) RelativePath() string {
 // ProjectName generates a name for the project that can be used
 // in the Infracost config file.
 func (p *Parser) ProjectName() string {
+	if p.projectName != "" {
+		return p.projectName
+	}
+
 	name := config.CleanProjectName(p.RelativePath())
 
 	if p.moduleSuffix != "" {
 		name = fmt.Sprintf("%s-%s", name, p.moduleSuffix)
+	} else if p.workspaceName != "" && p.workspaceName != defaultTerraformWorkspaceName {
```

Comment:
Added this so it could be more consistent with how `hcl_provider.ProjectName` worked.